### PR TITLE
Fix Matpower phase shifter import

### PR DESCRIPTION
**Pull request by geofjamg**
Signed-off-by: Geoffroy Jamgotchian <geoffroy.jamgotchian@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Comment by geofjamg** [comment]
@annetill for this PR we have to:
 - check if angle sign is correct (I did random)
 - add test
 - decide if we process the minAngle and maxAngle (for instance, create 3 taps, if defined)

File: matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerImporter.java [hunk 1]
```diff
@@ -269,6 +272,19 @@ private static void createBranches(MatpowerModel model, ContainersMapping contai
                         .setG(0)
                         .setB(mBranch.getB() / zb)
                         .add();
+                if (mBranch.getPhaseShiftAngle() != 0) {
```
Comment (annetill) [inline matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerImporter.java:275]: According to @zamarrenolm, the convention of angle increment sign has changed at version 3.2 of matpower files. After the version 3.2, the convention is the same as PSSE file:
`
    private static void tapChangerToPhaseTapChanger(TapChanger tapChanger, PhaseTapChangerAdder ptc) {
        ptc.setLowTapPosition(0)
            .setTapPosition(tapChanger.getTapPosition());

        tapChanger.getSteps().forEach(step ->
            ptc.beginStep()
                .setRho(1 / step.getRatio())
                .setAlpha(-step.getAngle())
                .setR(step.getR())
                .setX(step.getX())
                .setB(step.getB1())
                .setG(step.getG1())
                .endStep());
        ptc.add();
    }
`
And before, it was just the opposite :-)

Comment (marqueslanauja) [reply matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerImporter.java:275]: I generated a small case in `matpower .M` format (ascii format) starting from the `IEEE14` case and by changing a transformer to a phase shifter.
I solved the case on matpower and I exported the results as a `matpower .mat` format file.
Then I imported this file on this branch and I exported to `xiidm` format.
I ran the obtained `xiidm` case on `HELM` and on `open-Loadflow` obtaining the same result but different from the matpower one.
Then I modify the code to 

```
                    newTwt.newPhaseTapChanger()
                            .setTapPosition(0)
                            .beginStep()
                            .setRho(1)
                            .setAlpha(-mBranch.getPhaseShiftAngle())
                            .setR(0)
                            .setX(0)
                            .setG(0)
                            .setB(0)
                            .endStep()
                            .add();

```
After the modificación all three results are identical.
So the `alpha `attribute should be the opposite of the `mBranch.getPhaseShiftAngle()`.
The change is in concordance with the matpower documentation.

Comment (annetill) [reply matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerImporter.java:282]: Thanks a lot for your analysis, and your test makes us confident for the implementation. I have made the fix and I have added a fictitious test based on IEEE 14 test case. Just tell me if you agree with my modifications.

Comment (marqueslanauja) [reply matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerImporter.java:282]: I agree with your modification.
You added a new test that uses the resource `ieee14-phase-shifter.json` but I cannot find this resource. Maybe you have to commit it?
If you add the resource I will also check the test.

Comment (annetill) [reply matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerImporter.java:282]: Done, thank you it was a mistake!

Comment (marqueslanauja) [reply matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerImporter.java:282]: The test case works fine. I ran the `ieee14-phase-shifter` case on matpower and I checked that we get the same result on HELM and open-Loadflow. 
I will prepare an additional commit to update the voltage and angle attributes of the  `ieee14-phase-shifter.json` and `ieee14-phase-shiftermat.xiidm` resources with the values I got after solving the case on matpower. Do you agree?

Comment (geofjamg) [reply matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerImporter.java:282]: > According to @zamarrenolm, the convention of angle increment sign has changed at version 3.2 of matpower files. After the version 3.2, the convention is the same as PSSE file: ` private static void tapChangerToPhaseTapChanger(TapChanger tapChanger, PhaseTapChangerAdder ptc) { ptc.setLowTapPosition(0) .setTapPosition(tapChanger.getTapPosition());
> 
> ```
>     tapChanger.getSteps().forEach(step ->
>         ptc.beginStep()
>             .setRho(1 / step.getRatio())
>             .setAlpha(-step.getAngle())
>             .setR(step.getR())
>             .setX(step.getX())
>             .setB(step.getB1())
>             .setG(step.getG1())
>             .endStep());
>     ptc.add();
> }
> ```
> 
> ` And before, it was just the opposite :-)

According to https://matpower.org/docs/ref/matpower5.0/caseformat.html (and confirmed by our use cases), mat file version could be 1 or 2. @zamarrenolm @annetill are you sure about the version 3.2 as a breaking change version for phase shifter convention?

Comment (marqueslanauja) [reply matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerImporter.java:282]: Version 3.2 refers to MATPOWER software, not to file format. As you commented there are two versions of the MATPOWER file format (version 1 and version 2). 
MATPOWER software versions 3.2 and subsequent  use version 2 format. 
MATPOWER software versions 3.0 and earlier use version 1 format. 
Here is the MATPOWER software release history:
```
Pre 1.0, released Jun 25, 1997
Version 1.0, released Sep 17, 1997
Version 1.0.1, released Sep 19, 1997
Version 2.0, released Dec 24, 1997
Version 3.0, released Feb 14, 2005
Version 3.2, released Sep 21, 2007
Version 4.0, released Feb 7, 2011
Version 4.1, released Dec 14, 2011
Version 5.0, released  Dec 17, 2014
Version 5.1, released Mar 20, 2015
Version 6.0, released Dec 16, 2016
Version 7.0, released Jun 20, 2019
Version 7.1, released Oct 8, 2020
```

The sign convention used for phase shifters has been changed in version 3.2 to be consistent with PTI.

**Review by marqueslanauja — APPROVED** [review]
It is ok. 
Only a minor thing, we can update the` ieee14-phase-shifter.json` and `ieee14-phase-shiftermat.xiidm` resources by setting the MATPOWER solution in voltage and angle attributes. I can do it if you are agree.

**Pull request merged**